powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/encryptor/ecies/ClientEciesEncryptor.java

```diff
@@ -0,0 +1,218 @@
+                sharedInfo2Base,
+                envelopeKey.getEphemeralKeyPublic(),
+                requestNonce,
+                requestTimestamp,
+                associatedData
+        );
+        // Once we have SharedInfo2 prepared, we can construct an encryptor.
+        final EciesEncryptor eciesEncryptor = new EciesEncryptor(envelopeKey, sharedInfo2);
+        // If everything is OK, then encrypt the data.
+        final EciesPayload eciesPayload = eciesEncryptor.encrypt(
+                data,
+                new EciesParameters(requestNonce, associatedData, requestTimestamp)
+        );
+        // Keep envelope key and nonce used for the request if protocol require use the same nonce also for the response.
+        this.envelopeKey = envelopeKey;
+        this.requestNonce = validator.isUseTimestamp() ? null : requestNonce;
+
+        return new EncryptedRequest(
+                Base64.getEncoder().encodeToString(eciesPayload.getCryptogram().getEphemeralPublicKey()),
+                Base64.getEncoder().encodeToString(eciesPayload.getCryptogram().getEncryptedData()),
+                Base64.getEncoder().encodeToString(eciesPayload.getCryptogram().getMac()),
+                validator.isUseNonceForRequest() ? Base64.getEncoder().encodeToString(requestNonce) : null,
+                requestTimestamp
+        );
+    }
+
+    @Override
+    public boolean canDecryptResponse() {
+        return this.envelopeKey != null;
+    }
+
+    @Override
+    public byte[] decryptResponse(EncryptedResponse response) throws EncryptorException {
+        if (!canDecryptResponse()) {
+            throw new EncryptorException("Encryptor is not ready for response decryption.");
+        }
+
+        // Validate and decode response payload
+        if (!validator.validateEncryptedResponse(response)) {
+            throw new EncryptorException("Invalid encrypted response object");
+        }
+
+        final byte[] mac = Base64.getDecoder().decode(response.getMac());
+        final byte[] encryptedData = Base64.getDecoder().decode(response.getEncryptedData());
+        final byte[] responseNonce = validator.isUseTimestamp() ? Base64.getDecoder().decode(response.getNonce()) : requestNonce;
+        final Long responseTimestamp = validator.isUseTimestamp() ? response.getTimestamp() : null;
+
+        // Build sharedInfo2 with parameters received from the request.
+        final byte[] sharedInfo2 = EciesUtils.deriveSharedInfo2(
+                encryptorParameters.getProtocolVersion(),
+                sharedInfo2Base,
+                null,
+                responseNonce,
+                responseTimestamp,
+                associatedData
+        );
+        // Build decryptor object.
+        final EciesDecryptor decryptor = new EciesDecryptor(envelopeKey, sharedInfo2);
+        // Prepare EciesPayload and try to decrypt data.
+        final byte[] plaintext = decryptor.decrypt(new EciesPayload(
```

Comment:
This code is hard to read, can we please separate this code into multiple lines and introduce 2 variables for new objects?